### ISSUE TITLE
Is it possible to fetch a custom Hamiltonian (two-particle integral) into PySCF for unrestricted case?

### ISSUE THREAD

**Issue by yapolyak** [pyscf/pyscf#941]
Hi, 

It is possible to run restricted-orbital calculations for a custom Hamiltonian - but is there a way to do the same for unrestricted case? I couldn't find what shape should `_eri `object take in such a case - usually one deals with 4 (or 3) blocks of two-electron integrals with different spin symmetries: `aaaa`, `bbbb`, `aabb` (and `bbaa`), but there is only a single `_eri` object, which only seems to accept 4-D tensor? 

Thank you very much in advance!
Iakov

**Comment by sunqm** [comment]
This is not supported. For mean field object, you have to overwrite the get_jk or get_veff method. For post HF modules, you need to modify the ao2mo method.

**Comment by yapolyak** [comment]
Thank you very much!

**Comment by davidzsoltmanrique** [comment]
Hi All, I am just wondering, is this feature supported in the PySCF 2.0 and is there an example of its usage? Many thanks!

**Comment by yapolyak** [comment]
Yes, in the release notes for 2.0.0 you mention "Custom Hamiltonian for UHF", however it doesn't seem like `mf.get_hcore()` or `mf._eri()` now accept different spin symmetry blocks. I also couldn't find an example for the above new feature. Could you please kindly point to such an example if it exists so that we could better understand how to use this feature?

**Comment by yapolyak** [comment]
Just to update, I have found the relevant (I think) PR (https://github.com/pyscf/pyscf/pull/927) so I guess the idea is to use `init_guess_by_minao()`, right?
